doc: expand fs.watch caveats (#32176)
nodejs/node: doc/api/fs.md

An `ERR_FEATURE_UNAVAILABLE_ON_PLATFORM` exception will be thrown
when the option is used on a platform that does not support it.

> On Windows no events will be emitted if the watched directory is moved or

Reviewer: ```suggestion
On Windows, no events will be emitted if the watched directory is moved or
```

> renamed. If the watched directory is deleted `EPERM` error will be reported.

Reviewer 2: ```suggestion
renamed. An `EPERM` error is reported when the watched directory is deleted.
```
(but certainly `s/will be/is/` :-))
